### ISSUE TITLE
Windows 8.1 Universal on Phone - Graphics Glitch on Startup

### ISSUE THREAD

**Issue by jakepoz** [MonoGame/MonoGame#3948]
This bug occurs with Windows 8.1 Universal projects and the default Monogame templates running the latest nightly build (6/18/15, as well as 3.4 release builds).

We notice that very often on app startup, there is perhaps half a second where the graphics do not take up the full screen. This happens both on the Windows Phone Emulator and on a Lumia 920 test device. We do not see it when creating Windows Phone 8.0 projects, or when running on the desktop.

You can see it very clearly in this video:
https://www.youtube.com/watch?v=OAWHxEiTdnw

As the app starts up, it first renders to the top half of the screen. After half a second, rendering starts happening like normal.

Please let me know if any other information is needed to get a repro of this issue.

**Comment by tomspilman** [comment]
I've seen this too.  What is interesting is that the stock DirectX C++ sample for Win10 has the same behavior.

This makes me suspect it is a bug in the early builds of Win10 on phone.  I've mostly been waiting for Win10 to get finalized so we can make another pass fixing issues for it.

**Comment by jakepoz** [comment]
Yes, I've seen this on Windows 10, but it does happen on production builds of Windows Phone 8.1 that are live today. Do you think we should look into it on WP 8.1 first, and then come back to Win10 later?

**Comment by tomspilman** [comment]
Oh... totally missed the Win 8.1 there.  I can't say I know what is going on with it on that platform.

**Comment by jakepoz** [comment]
Doing a bit of digging here, I believe this is related to SupportedOrientations and their initial states.

If you comment out the line shown in the sample:
```C#
graphics.SupportedOrientations = DisplayOrientation.LandscapeLeft | DisplayOrientation.LandscapeRight;
```

and you run the template project starting in portrait mode, there is no graphics glitch. However, the graphics are scaled weird, because Monogame thinks your screen is 640x384 in size yet it scales the viewport to that. If you then rotate the device to landscape, the graphics device gets reset to look proper, and then if you rotate back again to portrait, the graphics device gets reset again and the portrait mode works properly too.

I believe what's happening is that on startup, we see a duplicate initialization of the graphics device, once for some strange 640x384 size and then again for whatever the proper size is.

**Comment by nkast** [comment]
This looks a lot like kakCAT video at http://community.monogame.net/t/wp8-1-game-freezes-sporadically/1865/17

My explanation is that the UI thread/message loop is slowed down by CompositionTarget.Rendering and it receive OrientationChanged event a few frames later. You can test this theory by set it in portrait mode, then run it , load IE and change to landscape, then resume your project to see it rendering wrong for a few frames before returning to portrait. Also look out for total freeze in WP8.1!

We have to render in another thread on W8.1, but that has a ton of problems/bugs. It's best to forget WP8.1 and wait for WP10 (I am not going to say W10 on phone...).

**Comment by tomspilman** [comment]
Fixed in #5520.
